src/environment.h

```diff
@@ -108,6 +112,21 @@ namespace librealsense
 
         int generate_stream_id() { return _stream_id.fetch_add(1); }
 
+        std::atomic< int > & get_device_power_counter( const std::string & uid )
+        {
+            return _uvc_device_to_power_counter[uid];
+        }
+
+        std::shared_ptr< platform::uvc_device > & get_uvc_device( const std::string & uvc_device_info_uid )
+        {
+            return _device_info_uid_to_uvc_device[uvc_device_info_uid];
+        }
+
+        void set_uvc_device( const std::string & uvc_device_info_uid, std::shared_ptr< platform::uvc_device > & device )
```

Comment:
This looks OK but the device is NEVER removed! I don't think that's what we want.
We want it to be removed when the last reference to it is gone.